includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1204,18 +1204,18 @@ public function get_intent_from_order( $order ) {
 	 * @param stdClass $intent The intent that is being processed.
 	 * @return bool            A flag that indicates whether the order is already locked.
 	 */
-	public function lock_order_payment( $order, $intent ) {
+	public function lock_order_payment( $order, $intent = null ) {
 		$order_id       = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $order->id : $order->get_id();
 		$transient_name = 'wc_stripe_processing_intent_' . $order_id;
 		$processing     = get_transient( $transient_name );
 
 		// Block the process if the same intent is already being handled.
-		if ( $processing === $intent->id ) {
+		if ( -1 == $processing || $processing === $intent->id ) {
```

Comment:
Since `$intent` can be `null`, we need to check if property `id` exists:

```
isset( $intent->id ) && $processing === $intent->id
```

That should fix both of the failing tests.

Also - is there a reason not to use triple equals for `-1 === $processing`?

Reply:
> Since $intent can be null, we need to check if property id exists:

Good catch, fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1083/commits/7214ca575dd0064e4d0676e597ae343280b187f3.

> Also - is there a reason not to use triple equals for -1 === $processing?

It was because I don't 100% trust types in options, but I decided to do that, so I switched to `'-1'`. Even though I was setting the transient to an integer, it was read as a string 🤷‍♂